apiserver/Makefile

```diff
@@ -122,7 +122,7 @@ KIND ?= $(REPO_ROOT_BIN)/kind
 KUSTOMIZE_VERSION ?= v3.8.7
 GOFUMPT_VERSION ?= v0.3.1
 GOIMPORTS_VERSION ?= latest
-GOLANGCI_LINT_VERSION ?= v1.50.1
+GOLANGCI_LINT_VERSION ?= v1.54.1
 KIND_VERSION ?= v0.19.0
```

Comment:
Do we have to put here a specific version? Can we just use the latest? Or figure out the version based on the go version? I am a bit concerned with the maintenance of updating the version constantly.

Reply:
No. I think we should pin the version as it will give us consistent builds. The maintenance is small price to pay.

Reply:
Agreed that we should pin the verison